cicd/crabserver_pypi/Dockerfile

```diff
@@ -0,0 +1,59 @@
+# Get CERN tsanames.ora from cmsweb rpm image
+FROM registry.cern.ch/cmsweb/cmsweb:20230427-stable as tnsnames
+
+# Use `wmagent-base` image as our baseimage:
+# https://github.com/dmwm/CMSKubernetes/blob/2b0454f9205cb8f97fecb91bf6661b59e4b31424/docker/pypi/wmagent-base/Dockerfile
+
+FROM registry.cern.ch/cmsweb/wmagent-base:pypi-20230705
+SHELL ["/bin/bash", "-c"]
+ENV WDIR=/data
+ENV USER=crab3
+RUN useradd ${USER} \
+    && install -o ${USER} -d ${WDIR} \
+    && echo "${USER} ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/01-crab3
+
+# install packages from debian official repo
+RUN apt-get update \
+    && apt-get install -y tini \
+    && apt-get clean all
+
+# Copy CERN tnsnames.ora
+COPY --from=tnsnames /etc/tnsnames.ora /etc/tnsnames.ora
+
+#RUN mkdir -p /opt/rucio/etc/
+
+RUN mkdir /build
+WORKDIR /build
+COPY cicd/crabserver_pypi/ .
+# Install wmcore/crab dependencies
+RUN pip install -r requirements.txt
+# Getting the version number from wmcore_requirements.txt and install it with `--no-deps`.
+# When reading from file, `grep -v '^\s*#'` to ignore all comment lines, including the ones with leading whitespace ('\s' to match whitespace).
+RUN req=($(grep -v '^\s*#' < wmcore_requirements.txt)) \
```

Comment:
this bash is still too much advanced for me to understand. I can run it and see that it works, but ($(cmd)) it is out of my league. Oh well...
I guess you do not like using `cut` or `awk` to parse the two elements of `https://github.com/dmwm/WMCore 2.2.4rc6`.

Reply:
It is the bash built-in on how you create array (put string separate by space inside parenthesis and assign to variable).

`cut` is also fine for me. What about 
```bash
wmcore_version=$(grep -v '^\s*#' wmcore_requirements.txt | cut -d' ' -f2)
```
?